src/components/input/field-value.ts

```diff
@@ -0,0 +1,32 @@
+// source: https://github.com/angular/material2/blob/master/src/lib/core/annotations/field-value.ts
+
+/**
+ * Annotation Factory that allows HTML style boolean attributes. For example,
+ * a field declared like this:
+ * @Directive({ selector: 'component' }) class MyComponent {
+ *   @Input() @BooleanFieldValueFactory() myField: boolean;
+ * }
+ *
+ * You could set it up this way:
+ *   <component myField>
+ * or:
+ *   <component myField="">
+ * @deprecated
+ */
+function booleanFieldValueFactory() {
+  return function booleanFieldValueMetadata(target: any, key: string): void {
+    const defaultValue = target[key];
+    const localKey = `__md_private_symbol_${key}`;
+    target[localKey] = defaultValue;
+
+    Object.defineProperty(target, key, {
+      get() { return (<any>this)[localKey]; },
+      set(value: boolean) {
+        (<any>this)[localKey] = value != null && `${value}` !== 'false';
+      }
+    });
+  };
+}
+
+
+export { booleanFieldValueFactory as BooleanFieldValue };
```

Comment:
Hi, thanks for the PR, I was thinking about doing something very similar so time ago!

We do not use this syntax too much in the framework, could you replace it by changing:

```ts
function booleanFieldValueFactory() {
```
to:
```ts
export function BooleanFieldValue() {
```


Thanks!